pymodbus/transaction.py

```diff
@@ -420,8 +423,10 @@ def _recv(self, expected_response_length, full) -> bytes:  # noqa: C901
                 min_size = 4
             elif isinstance(self.client.framer, ModbusAsciiFramer):
                 min_size = 5
-            else:
+            elif expected_response_length:
```

Comment:
This do not seem right, however it is not covered by tests.

Reply:
This was flagged by the type-checker, if `expected_response_length` is none we can't assign `min_size` to `expected_response_length` as we pass `min_size` to `self.client.framer.recvPacket` which expects an integer.

Reply:
Seems you need more typing.

Reply:
Apparently tweaking the style here a bit using original behavior got it passing the type checker.